mesop/components/uploader/uploader.py

```diff
@@ -0,0 +1,72 @@
+import json
+from dataclasses import dataclass
+from typing import Any, Callable
+
+import mesop.components.uploader.uploader_pb2 as uploader_pb
+from mesop.component_helpers import (
+  insert_component,
+  register_event_handler,
+  register_event_mapper,
+  register_native_component,
+)
+from mesop.events import MesopEvent
+
+
+@dataclass(kw_only=True)
+class UploadChangeEvent(MesopEvent):
+  """Event representing a change in the uploader component's value.
+
+  Attributes:
+      name: The file's name.
+      size: The size of the file in bytes.
+      type: The file's MIME type.
+      data_url: Uploaded content as data URL.
+      key (str): key of the component that emitted this event.
+  """
+
+  name: str
+  size: int
+  type: str
+  data_url: str
+
+
+register_event_mapper(
+  UploadChangeEvent,
+  lambda event, key: UploadChangeEvent(
+    key=key.key, **json.loads(event.string_value)
+  ),
+)
+
+
+@register_native_component
+def uploader(
+  *,
+  key: str | None = None,
+  accept: str = "",
```

Comment:
I see you're following the native web API here and this is where I think it may make sense to deviate a little bit to make it more pythonic. IIUC,  `accept` is a comma-separated list-like string (for whatever reason HTML attributes don't really have a native list type). I think having this be an actual list[str] is more pythonic and then naming it something like `accept_types` makes it more obvious this is a list and not a str.

As reference:
- https://docs.streamlit.io/develop/api-reference/widgets/st.file_uploader
- https://www.gradio.app/docs/uploadbutton

Reply:
Good point. I agree that that is a better API.